File: JS/No_Framework/Factory/js/factory.js
// Character Factory
class CharacterFactory {

  // create spy, programmer, engineer and default
  static createCharacter(...args) {

    return new Promise((resolve, reject) => {

      switch (args[0]) {
        case 'spy': resolve(new Spy(...args));
        case 'programmer': resolve(new Programmer(...args));
        case 'engineer': resolve(new Engineer(...args));
        default: reject('Can not create this character');
      }
    });
  }
}

File: Python/factory_demo/factory.py
# import
from concrete import Spy, Programmer, Engineer


# --- Character Factory ---
class CharacterFactory:

    # instansiate character type
    @staticmethod
    def create_character(char_type, char_name, char_weapon):

        if char_type == 'spy':
            return Spy(char_type, char_name, char_weapon)

        elif char_type == 'programmer':
            return Programmer(char_type, char_name, char_weapon)

        elif char_type == 'engineer':
            return Engineer(char_type, char_name, char_weapon)

        else:
            print('Can not create this character.')
